sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/http/rest/RestProxyBase.java

```diff
@@ -71,10 +69,10 @@ public final Object invoke(Object proxy, final Method method, RequestOptions opt
             context = RestProxyUtils.mergeRequestOptionsContext(context, options);
 
             request.getMetadata().setContext(context);
-            request.getMetadata().setRequestLogger(methodParser.getMethodLogger());
-            request.getMetadata().setEagerlyConvertHeaders(methodParser.isHeadersEagerlyConverted());
-            request.getMetadata().setEagerlyReadResponse(methodParser.isResponseEagerlyRead());
-            request.getMetadata().setIgnoreResponseBody(methodParser.isResponseBodyIgnored());
+            request.getMetadata().setRequestLogger(methodParser.getMethodLogger());;
+            request.getMetadata().setResponseBodyHandling(options != null
```

Comment:
Need to check `options.getResponseBodyHandling` for being null as well.

Reply:
I had it set to `BUFFER` by default in both `RequestOptions` and `HttpRequestMetadata`. Should we not initialize it instead and let it be figured out by `SwaggerMethodParser`?

Reply:
I think I'll do that :P